lib/active_model/serializer/adapter/json.rb

```diff
@@ -2,10 +2,15 @@ class ActiveModel::Serializer::Adapter::Json < ActiveModel::Serializer::Adapter
         extend ActiveSupport::Autoload
         autoload :FragmentCache
 
+        def initialize(serializer, options = {})
+          super
+          @included = ActiveModel::Serializer::Utils.include_args_to_hash(@options[:include] || '*')
+        end
+
         def serializable_hash(options = nil)
           options ||= {}
           if serializer.respond_to?(:each)
-            result = serializer.map { |s| FlattenJson.new(s).serializable_hash(options) }
+            result = serializer.map { |s| FlattenJson.new(s, include: @included).serializable_hash(options) }
```

Comment:
The name `expand_includes` was poorly chosen. What it does is take an "include hash" and spits out an array of pairs `[Association, RelatedIncludeHash]`, that's why it's the serializer's job in my opinion.